src/WingetCreateCLI/UserSettings.cs

```diff
@@ -0,0 +1,170 @@
+    using System.IO;
+    using Microsoft.WingetCreateCLI.Logging;
+    using Microsoft.WingetCreateCLI.Models.Settings;
+    using Microsoft.WingetCreateCLI.Properties;
+    using Newtonsoft.Json;
+    using Sharprompt;
+
+    /// <summary>
+    /// UserSettings configuration class for WingetCreate.
+    /// </summary>
+    public static class UserSettings
+    {
+        private static readonly Lazy<string> SettingsPathLazy = new(() => Path.Combine(Common.LocalAppStatePath, "settings.json"));
+
+        private static readonly Lazy<string> SettingsBackupPathLazy = new(() => Path.Combine(Common.LocalAppStatePath, "settings.json.backup"));
+
+        static UserSettings()
+        {
+            LoadSettings();
+        }
+
+        /// <summary>
+        /// Gets the path to the settings json file.
+        /// </summary>
+        public static string SettingsJsonPath => SettingsPathLazy.Value;
+
+        /// <summary>
+        /// Gets the path to the backup settings json file.
+        /// </summary>
+        public static string SettingsBackupJsonPath => SettingsBackupPathLazy.Value;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to disable telemetry.
+        /// </summary>
+        public static bool TelemetryDisable
+        {
+            get
+            {
+                return Settings.Telemetry.Disable;
+            }
+
+            set
+            {
+                Settings.Telemetry.Disable = value;
+                SaveSettings();
+            }
+        }
+
+        private static SettingsManifest Settings { get; set; }
+
+        /// <summary>
+        /// Attempts to parse a settings json file to determine if the file is valid based on the settings schema.
+        /// </summary>
+        /// <param name="path">Path to settings json file.</param>
+        /// <returns>A boolean IsValid that indicates if the file was parsed successfully and a list of error strings if the parsing fails.</returns>
+        public static (bool IsValid, List<string> Errors) ParseJsonFile(string path)
+        {
+            SettingsManifest settingsManifest;
+            List<string> errors = new List<string>();
+
+            using (StreamReader r = new StreamReader(path))
+            {
+                string json = r.ReadToEnd();
+                settingsManifest = JsonConvert.DeserializeObject<SettingsManifest>(
+                    json,
+                    new JsonSerializerSettings
+                    {
+                        Error = (sender, args) =>
+                        {
+                            errors.Add(args.ErrorContext.Error.Message);
+                            args.ErrorContext.Handled = true;
+                        },
+                    });
+            }
+
+            if (settingsManifest == null)
+            {
+                errors.Add("Manifest cannot be empty.");
+            }
+
+            bool isValid = errors.Count == 0 && settingsManifest != null;
```

Comment:
> && settingsManifest != null [](http://example.com/codeflow?start=44&length=28)

redundant

Reply:
removed null check, now only checks that errors.Count is equal to 0